utils_test.go

```diff
@@ -42,3 +50,202 @@ func TestValidStoragePathFormat(t *testing.T) {
 		assert.NilError(t, err)
 	}
 }
+
+func TestRootlessRuntimeDir(t *testing.T) {
+	tmpDir, err := ioutil.TempDir(os.TempDir(), "containers-user-*")
+	defer os.RemoveAll(tmpDir)
+	assert.NilError(t, err)
+
+	// testing homedir.GetRuntimeDir binding.
+
+	monkey.Patch(homedir.GetRuntimeDir, func() (string, error) {
```

Comment:
I guess this all can work, but the implementation of monkey patching is fairly scary (`mprotect` to RWX, is that even allowed with SELinux, and will it continue to be?) — and mostly not necessary.

In this case, `os.Setenv("XDG_RUNTIME_DIR")`

Reply:
The license for the module also looks... incompatible.

Reply:
Yes, but I can't find any other tool for monkey patching for go. It is not possible to mock system calls (as I know) without monkey patching.

Reply:
I will try to write parameters version of function to test it without monkey patching.